### PR TITLE
Calling automation - config screen

### PR DESCRIPTION
**Pull request by anjulgarg**
# What
Test for local config screen validation

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-composites/tests/browser/call/fixture.ts [hunk 1]
```diff
@@ -52,8 +52,16 @@ export const test = base.extend<unknown, ChatWorkerFixtures>({
...
-        args: ['--start-maximized', '--disable-features=site-per-process'],
-        headless: false
+        executablePath: process.env.CHROME_PATH,
+        args: [
+          '--start-maximized',
+          '--disable-gpu',
+          '--use-fake-ui-for-media-stream',
+          '--use-fake-device-for-media-stream',
+          '--allow-file-access',
+          `--use-file-for-fake-video-capture=${path.join(__dirname, 'puppy.y4m')}`
```
Comment (JamesBurnside) [inline packages/react-composites/tests/browser/call/fixture.ts:62]: To help with file size, sounds like we can use mpjeg also? (if its more compressed?)
https://source.chromium.org/chromium/chromium/src/+/master:media/capture/video/file_video_capture_device.h;l=33

Comment (JamesBurnside) [reply packages/react-composites/tests/browser/call/fixture.ts:62]: If we use a super small file that is literally one frame, what happens? (does it just loop continuosuly?)

Comment (JamesBurnside) [reply packages/react-composites/tests/browser/call/fixture.ts:62]: From that link, is just JPEG supported?? 
image

File: packages/react-composites/tests/browser/call/CallComposite.test.ts [hunk 2]
```diff
@@ -12,7 +27,19 @@ test.describe('Call Composite E2E Tests', () => {
       await pages[idx].waitForSelector(dataUiId('call-composite-device-settings'));
       await pages[idx].waitForSelector(dataUiId('call-composite-local-preview'));
       await pages[idx].waitForSelector(`${dataUiId('call-composite-start-call-button')}[data-is-focusable="true"]`);
-      expect(await pages[idx].screenshot()).toMatchSnapshot(`page-${idx}-call-screen.png`);
+      await stubLocalCameraName(pages[idx]);
+      expect(await pages[idx].screenshot()).toMatchSnapshot(`page-${idx}-call-screen.png`, { threshold: 0.5 });
```
Comment (JamesBurnside) [inline packages/react-composites/tests/browser/call/CallComposite.test.ts:31]: for threshold, is this something we decide per test or will we apply .5 to all tests? (what does 0.5 really mean?)

Comment (JamesBurnside) [reply packages/react-composites/tests/browser/call/CallComposite.test.ts:31]: Ah looks to be a guess and hope value 😅 https://github.com/mapbox/pixelmatch#api
What was venturing beyond the threshold in these tests?

File: packages/react-composites/tests/browser/call/fixture.ts (hunk 1 above)
Comment (anjulgarg) [reply packages/react-composites/tests/browser/call/fixture.ts:62]: Did try to use MJPEG and JPEG but it didn't work. I will try again today.

**Review by alcail — APPROVED** [review]

File: packages/react-composites/tests/browser/call/fixture.ts (hunk 1 above)
Comment (anjulgarg) [reply packages/react-composites/tests/browser/call/fixture.ts:64]: So JPEG and MJPEG is not working.
But I found a .y4m file in chromium repository that's just 744kb and is small enough to be used in our tests.

Comment (JamesBurnside) [reply packages/react-composites/tests/browser/call/fixture.ts:64]: Sounds good to me!

Comment (JamesBurnside) [reply packages/react-composites/tests/browser/call/fixture.ts:64]: (assuming no licensing/copyright issues from out using theirs)

File: packages/react-composites/tests/browser/utils.ts [hunk 3]
```diff
@@ -11,7 +11,7 @@ export const dataUiId = (v: string): string => `[${DATA_UI_ID}="${v}"]`;
 const DATA_UI_ID = 'data-ui-id';
 const CONNECTION_STRING = process.env.CONNECTION_STRING ?? '';
 const PAGE_VIEWPORT = {
```
Comment (JamesBurnside) [inline packages/react-composites/tests/browser/utils.ts:13]: can you leave a comment here that this should match the value set in the fixture.ts (if it needs to), or use the same value from a variable from a const file

**Review by JamesBurnside — APPROVED** [review]

File: packages/react-composites/tests/browser/call/fixture.ts (hunk 1 above)
Comment (anjulgarg) [reply packages/react-composites/tests/browser/call/fixture.ts:64]: It's just 4 colored squares :D

File: packages/react-composites/tests/browser/call/CallComposite.test.ts (hunk 2 above)
Comment (anjulgarg) [reply packages/react-composites/tests/browser/call/CallComposite.test.ts]: Tweaked the code and it doesn't have a need for threshold. (For now)

**Pull request merged**